src/thing/remove.ts

```diff
@@ -64,13 +60,14 @@ export function removeAll(thing: Thing, property: Url | UrlString): Thing {
   if (!internal_isValidUrl(property)) {
     throw new ValidPropertyUrlExpectedError(property);
   }
-  const predicateNode = asNamedNode(property);
+  const predicateIri = internal_toIriString(property);
+  const newPredicates = { ...thing.predicates };
+  delete newPredicates[predicateIri];
```

Comment:
Hurray for efficient and compact removal ! (Not the only place, but worth mentioning somewhere)